Packages/com.unity.inputsystem/InputSystem/Plugins/XR/TrackedPoseDriver.cs

```diff
@@ -320,19 +354,40 @@ protected virtual void OnBeforeRender()
             }
         }
 
-        protected virtual void SetLocalTransform(Vector3 newPosition, Quaternion newRotation)
+        protected virtual void SetLocalTransform(Vector3 newPosition, Quaternion newRotation, bool hasTrackingStateFlagInfo, PoseDataFlags trackingStateFlags)
```

Comment:
This method signature doesn't look like it has to change since you are using member variables to track `m_HasTrackingStateFlags` and `m_CurrentTrackingStateFlags`, correct? Are there other consumers of this function that will change the transform data without changing the 'new' internal state of this object?

Reply:
Two things:
* Can't `hasTrackingStateInfo` be derived from `trackingStateFlags`? (eg, `hasTrackingStateInfo = trackingStateFlags != PoseDataFlags.None;`
* You could also just overload the method and have to 2-param version call into the 4-param version (using the member variables @vrdave-unity alluded to above to supply the additional parameters?)

Reply:
added a pass through.

and i need the additional flag because i want tracking to work if there's no action providing tracking state flags. a None will cause nothing to be applied.